internal/app/registry.go

```diff
@@ -0,0 +1,60 @@
+package app
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/asynkron/protoactor-go/actor"
+)
+
+var errAlreadyRegistered = errors.New("factory already registered")
+
+var (
+	registry Registry
+	rwMu     sync.RWMutex
+)
+
+type (
+	Registry    map[string]FactoryFunc
+	FactoryFunc func(any) actor.Producer
+)
```

Comment:
I fully agree on adding the documentation and will proceed as suggested.

However, I don’t think adding a specific type for configuration is relevant. The type is set to `any` because the controller acts as a transparent pass-through, providing configurations to applications without interference. And it’s up to each application to validate its own configurations.